plugins/csharp/Managed/VideoSource.cs

```diff
@@ -12,62 +12,91 @@ public static class VideoSource {
         private static Bitmap PendingBitmap;
         private static Bitmap CurrentBitmap;
         private static String Result;
+        private static DateTime LastRequestTime;
 
-        private static uint IngestedImageCount;
-        private static uint AwaitedImageCount;
-        private static uint ResultCount;
-        private static uint StateFlag;
         private static bool WaitingForResult;
 
+        public static int Timeout { get; set; }
+
         static VideoSource() {
             Lock = new object();
+            LastRequestTime = DateTime.Now;
+            Timeout = 1000;
         }
 
         public static string SendImage(Bitmap bitmap) {
             lock (Lock) {
                 PendingBitmap = bitmap;
-                IngestedImageCount++;
-                WaitingForResult = true;
-                Monitor.Pulse(Lock);
+                Console.WriteLine($"THREAD {Thread.CurrentThread.ManagedThreadId} SENT IMAGE");
```

Comment:
Can we do better for debug log? Less spammy, or at least more control as to the level of spamminess?